provider/scripts/spark/offline_store_spark_runner.py

```diff
@@ -742,6 +774,24 @@ def parse_args(args=None):
     arguments.spark_config = split_key_value(arguments.spark_config)
     arguments.credential = split_key_value(arguments.credential)
 
+    if (
+        arguments.transformation_type == "sql"
+        and arguments.job_type == JobType.BATCH_FEATURES
+        and arguments.submit_params_uri is not None
+    ):
+        if arguments.store_type == "s3":
```

Comment:
this is being picky but I am wondering if it makes sense to just have a function that downloads the file locally or as bytes and then we can parse it? 

Then the function can be used for pulling the df transformation too

Reply:
This needs to be added for the rest of the filestores, so we should try to make things a bit more reusable/generic in there

Reply:
So, I thought the same thing given I had to pull `get_s3_object` out of `get_code_from_file` which only handles pickle files, but I decided I wouldn't undertake refactoring this login in this PR.